src/ssc_joystick_ros.cpp

```diff
@@ -564,19 +598,19 @@ int main(int argc, char **argv)
       // Joystick has timed out
       std::cout << "JOYSTICK TIMEOUT" << std::endl;
       last_joystick_msg = now_sec;
-      engaged = 0;
+      engaged = speed_engaged = steering_engaged = 0;
     }
 
     // Send output messages
     speed_msg.header.stamp = now;
-    speed_msg.mode = engaged;
+    speed_msg.mode = speed_engaged;
```

Comment:
just change this to: speed_msg.mode = (engage_speed_module > 0) ? engaged : 0;
Do the same for steering, then you can get rid of the speed_engaged and steering_engaged flags.  How you have it now you are only setting them to 1 before the loop starts, so I don't see how this could be working as is.